### DISCUSSION TITLE
How do I cache the .next/cache folder in github actions while building with docker

### DISCUSSION THREAD

**Discussion by mihir-karbelkar-wiz** [vercel/next.js#52577] [Help]
### Summary

We use github actions with docker but our builds are going above 10 minutes, however locally if I have the .next/cache folder the consecutive builds are much faster.

How can I do the same with docker, once docker build is done how can I export the .next/cache folder so that I can pull it the next time.

### Additional information

```js
action file is something like
docker build & docker push

docker file is three staged for dependencies, building and the running.
```

**Comment by nabildroid** [comment]
you can save the .next/cache in cloud storage for example google cloud storage and in the next build you download the file again and use it in building the docker

**Comment by nabildroid** [comment]
```yaml
# Use a Node.js base image
FROM node:14 AS build

# Set the working directory
WORKDIR /app

# Copy the package.json and yarn.lock files
COPY package.json yarn.lock ./

# Install dependencies
RUN yarn install

# Copy the rest of the application code
COPY . .

# Build the Next.js site
RUN yarn build

# Save the generated .next/cache to Google Cloud Storage
RUN gsutil -m cp -r .next/cache gs://your-bucket/.next/cache

# Use a new Docker stage for the subsequent build
FROM node:14 AS final
```
